### ISSUE TITLE
Noninteractive mode still requires user interaction

### ISSUE THREAD

**Issue by mbooth101** [flatpak/flatpak#3848]
## Linux distribution and version

Fedora 32

## Flatpak version

flatpak-1.6.5-1.fc32.x86_64

## Description of the problem

When run in noninteractive mode, the flatpak command still requires user interaction in some cases

## Steps to reproduce

For example:
```
  $ flatpak install --noninteractive --assumeyes  flathub org.videolan.VLC.Plugin.bdj
  Similar refs found for ‘org.videolan.VLC.Plugin.bdj’ in remote ‘flathub’ (system):
  
     1) runtime/org.videolan.VLC.Plugin.bdj/x86_64/3-19.08
     2) runtime/org.videolan.VLC.Plugin.bdj/x86_64/3-18.08
  
  Which do you want to use (0 to abort)? [0-2]: 
```

I would suggest that in such cases that flatpak should just install the newest version of the possible options. In this case ```runtime/org.videolan.VLC.Plugin.bdj/x86_64/3-19.08```

**Comment by mwleeds** [comment]
The current behavior if such a command is run without a tty attached is that it errors out. We probably should do that in the tty case too, better not to assume user intent when there's ambiguity.

**Comment by dbrgn** [comment]
> The current behavior if such a command is run without a tty attached is that it errors out. We probably should do that in the tty case too, better not to assume user intent when there's ambiguity.

I agree with this, I was confused when seeing an input prompt after using `--noninteractive`.

**Comment by vgdaut** [comment]
I suggest to throw an error in case of any ambiguity in non-interactive mode. If the user prohibits questions being asked, they must prevent any possible ambiguities. In scripting, this makes sense, as it is better if my script fails with an error message than it installs something that I did not want.

**Comment by LukeShortCloud** [comment]
Is there a short-term workaround for this?

I have tried various ways of using `echo` and `yes` to send standard input to Flatpak but nothing seems to be working. I am trying to install MangoHud non-interactively.

```
$ flatpak install -y --noninteractive org.freedesktop.Platform.VulkanLayer.MangoHud
Looking for matches…
Similar refs found for ‘org.freedesktop.Platform.VulkanLayer.MangoHud’ in remote ‘flathub’ (system):

   1) runtime/org.freedesktop.Platform.VulkanLayer.MangoHud/x86_64/21.08
   2) runtime/org.freedesktop.Platform.VulkanLayer.MangoHud/x86_64/22.08
   3) runtime/org.freedesktop.Platform.VulkanLayer.MangoHud/x86_64/23.08

Which do you want to use (0 to abort)? [0-3]:
```

**Comment by hfiguiere** [comment]
Specify the version. Otherwise it can't know which one you want.

**Comment by LukeShortCloud** [comment]
Ah, thank you. I can confirm that works. I just saw the keyword "runtime" and assumed it was a dependency.

Looking at Debian and Fedora package managers for inspiration, they always install the latest version of a package even if older packages are available in their repositories. If a user does not have it installed, it would be safe to assume they want the latest version. If they wanted an older version for any reason, they would most likely have that version already explicitly stated in their install command.

**Comment by smcv** [comment]
Runtime extensions are not apps, so there is no well-defined "latest version": they're more like plugins. If what you want is "the version of MangoHud that plugs into o.fd.Platform 22.08", then "the version of MangoHud that plugs into o.fd.Platform 23.08" is not an acceptable substitute. Flatpak can't know whether you are aiming to display MangoHud as an overlay over Steam (which currently uses o.fd.Platform 23.08), or as an overlay over some older game that still uses o.fd.Platform 22.08, or neither, or both.

So, yes you really do need to specify a version here, and Flatpak can't "just know" what you meant in this case.

**Comment by MrDrMcCoy** [comment]
I still think handling needs to be in place for non-interactive installs of runtime extensions without specifying a version. Not having this places a maintenance burden on install script maintainers and documentation writers for consuming these runtimes by their client apps whenever the app or runtime release new versions. Tracking that is not easy to do.

Perhaps we can include logic for selecting all applicable runtime versions in this case, which would map to all runtime versions that have an installed or incoming app Flatpak that can use it. This could be a no-op if the user asks to non-interactively install a runtime, but no app Flatpaks are installed or incoming that are able to use it.

If Flatpaks do not currently have metadata fields that would make this matching possible, I propose adding that in.
